src/index.js

```diff
@@ -0,0 +1,202 @@
+          <path style=" stroke:none;fill-rule:nonzero;fill:rgb(92.156863%,27.058824%,34.901961%);fill-opacity:1;" d="M 10.5 6.375 L 6.375 6.375 C 6.167969 6.375 6 6.207031 6 6 C 6 5.792969 6.167969 5.625 6.375 5.625 L 10.5 5.625 C 10.707031 5.625 10.875 5.792969 10.875 6 C 10.875 6.207031 10.707031 6.375 10.5 6.375 Z M 10.5 6.375 "/>
+          <path style=" stroke:none;fill-rule:nonzero;fill:rgb(92.156863%,27.058824%,34.901961%);fill-opacity:1;" d="M 6.375 7.875 L 5.25 7.875 C 5.042969 7.875 4.875 7.707031 4.875 7.5 C 4.875 7.292969 5.042969 7.125 5.25 7.125 L 6.375 7.125 C 6.582031 7.125 6.75 7.292969 6.75 7.5 C 6.75 7.707031 6.582031 7.875 6.375 7.875 Z M 6.375 7.875 "/>
+          <path style=" stroke:none;fill-rule:nonzero;fill:rgb(92.156863%,27.058824%,34.901961%);fill-opacity:1;" d="M 9.75 7.875 L 7.875 7.875 C 7.667969 7.875 7.5 7.707031 7.5 7.5 C 7.5 7.292969 7.667969 7.125 7.875 7.125 L 9.75 7.125 C 9.957031 7.125 10.125 7.292969 10.125 7.5 C 10.125 7.707031 9.957031 7.875 9.75 7.875 Z M 9.75 7.875 "/>
+          <path style=" stroke:none;fill-rule:nonzero;fill:rgb(92.156863%,27.058824%,34.901961%);fill-opacity:1;" d="M 5.25 17.25 L 4.125 17.25 C 3.917969 17.25 3.75 17.082031 3.75 16.875 C 3.75 16.667969 3.917969 16.5 4.125 16.5 L 5.25 16.5 C 5.457031 16.5 5.625 16.667969 5.625 16.875 C 5.625 17.082031 5.457031 17.25 5.25 17.25 Z M 5.25 17.25 "/>
+          <path style=" stroke:none;fill-rule:nonzero;fill:rgb(92.156863%,27.058824%,34.901961%);fill-opacity:1;" d="M 10.875 17.25 L 6.75 17.25 C 6.542969 17.25 6.375 17.082031 6.375 16.875 C 6.375 16.667969 6.542969 16.5 6.75 16.5 L 10.875 16.5 C 11.082031 16.5 11.25 16.667969 11.25 16.875 C 11.25 17.082031 11.082031 17.25 10.875 17.25 Z M 10.875 17.25 "/>
+          <path style=" stroke:none;fill-rule:nonzero;fill:rgb(92.156863%,27.058824%,34.901961%);fill-opacity:1;" d="M 6.75 18.75 L 5.625 18.75 C 5.417969 18.75 5.25 18.582031 5.25 18.375 C 5.25 18.167969 5.417969 18 5.625 18 L 6.75 18 C 6.957031 18 7.125 18.167969 7.125 18.375 C 7.125 18.582031 6.957031 18.75 6.75 18.75 Z M 6.75 18.75 "/>
+          <path style=" stroke:none;fill-rule:nonzero;fill:rgb(92.156863%,27.058824%,34.901961%);fill-opacity:1;" d="M 10.125 18.75 L 8.25 18.75 C 8.042969 18.75 7.875 18.582031 7.875 18.375 C 7.875 18.167969 8.042969 18 8.25 18 L 10.125 18 C 10.332031 18 10.5 18.167969 10.5 18.375 C 10.5 18.582031 10.332031 18.75 10.125 18.75 Z M 10.125 18.75 "/>
+        </g>
+      </svg>
+      <span class="">Wind:</span>
+    <p class="">Speed: ${data.wind.speed}</p>
+    <p class="">Degree: ${data.wind.deg}</p>
+    </p>
+  </div>
+</div>`;
+
+  const parent = document.getElementById('reports');
+  parent.innerHTML += div;
+};
+
+form.addEventListener('submit', async (e) => {
+  e.preventDefault();
+  showSpinner();
+  const vars = getVars(e);
+  if (checkVars(vars)) {
+    const { data, err } = await getRes(vars);
```

Comment:
- [ ] When I submit a city which is not found, the spinner is stuck and keeps loading even after the request has returned from the weather API. You can check the network tab after you submit a search for a city that doesn't exist.

I suggest that you use here `then` and `catch` blocks with getRes(vars):

```js
getRes(vars)
  .then((data) => {
      // handle successful response (city found)
  }).catch((error) => {
      // handle no city found case
      // showMsg
  };
  ```

Important Note: When you issue a search for a "not found" city, the fetch resolve the promise because the weather API sends back a response and it's considered a successful request-response even if the city is not found. The only case that fetch reject a promise when an error happens within fetch itself (network error, server doesn't respond, ...)

Kindly check out [this](https://stackoverflow.com/questions/39297345/fetch-resolves-even-if-404)

Reply:
That was very helpful, thank you so much. I hope I implemented it correctly ^^